### PR TITLE
Feat: add optional @limit macro

### PR DESCRIPTION
**Pull request by z3z1ma**
Adds limit macro. Pending an update in sqlglot to add the `skip_limit_token` kwarg.

### CODE REVIEW

**Comment by z3z1ma** [comment]
Depends on https://github.com/tobymao/sqlglot/pull/3149

File: sqlmesh/core/macros.py [hunk 1]
```diff
@@ -539,158 +539,30 @@ def filter_(evaluator: MacroEvaluator, *args: t.Any) -> t.List[t.Any]:
...
-    return expression if evaluator.eval_expression(condition) else None
-
-
-@macro()
-def order_by(
-    evaluator: MacroEvaluator,
-    condition: exp.Condition,
-    expression: exp.Order,
-) -> t.Optional[exp.Order]:
-    """Inserts ORDER BY expression when the condition is True
```
Comment (tobymao) [inline sqlmesh/core/macros.py:677]: i don't like how we lose documentation, any way we can keep these docs? perhaps in _optional expression?

Comment (z3z1ma) [reply sqlmesh/core/macros.py:677]: I added the examples back to the `_optional_expression`. These run on doc tests and show usage for the various permutations. Is this what you meant?

Comment (z3z1ma) [reply sqlmesh/core/macros.py:677]: also if its not obvious in the diff BTW, `_optional_expression` already had docs, so we werent losing docs but dropping strings that are 90% repeated.

Comment (tobymao) [reply sqlmesh/core/macros.py:677]: exactly

**Review by tobymao — APPROVED** [review]

**Pull request merged**